tests/utils/utils.go

```diff
@@ -1175,3 +1178,37 @@ func GetChainConfigWithOffChainMessages(offChainMessages []avalancheWarp.Unsigne
 
 	return string(offChainMessageJson)
 }
+
+// read in the template file, make the substitutions declared at the beginning
+// of the function, write out the instantiation to a temp file, and then return
+// the path to that temp file.
+func InstantiateGenesisTemplate(
+	templateFileName string,
+	chainID uint64,
+) string {
+	substitutions := []struct {
+		Target string
+		Value  string
+	}{
+		{
+			"<EVM_CHAIN_ID>",
+			strconv.FormatUint(chainID, 10),
+		},
+	}
+
+	templateFileBytes, err := os.ReadFile(templateFileName)
+	Expect(err).Should(BeNil())
+
+	subnetGenesisFile, err := os.CreateTemp(os.TempDir(), "")
+	defer subnetGenesisFile.Close()
+	Expect(err).Should(BeNil())
+
+	var replaced string = string(templateFileBytes[:])
+	for _, s := range substitutions {
+		replaced = strings.Replace(replaced, s.Target, s.Value, 1)
```

Comment:
It seems a little odd to generalize the target text replacement in this loop, but not the number of replacements. Since we're only replacing a single value in the genesis, I think it's appropriate to just explicitly call `strings.Replace(replaces, "<EVM_CHAIN_ID>",  strconv.FormatUint(chainID, 10))` here. Or we can go in the other direction and also generalize the number of replacements by either calling `strings.ReplaceAll` or putting `n` in the `substitutions` struct.